service/controller/chartconfig/v7/resource/chart/current_test.go

```diff
@@ -122,6 +123,23 @@ func Test_CurrentState(t *testing.T) {
 			returnedError:  fmt.Errorf("Unexpected error"),
 			expectedError:  true,
 		},
+		{
+			name: "case 4: chart cordoned",
+			obj: &v1alpha1.ChartConfig{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						"chart-operator.giantswarm.io/cordon-reason": "testing upgrade",
+						"chart-operator.giantswarm.io/cordon-until":  "2019-12-31T23:59:59Z",
```

Comment:
I think we should increment the year by like 100 so tests stay valid also in 2020 and after.

Reply:
If we have logic at now, it makes sense but we are going to have that feature in cluster-operator for alarming.